### PR TITLE
Various build fixes from LEXCI in-person meeting

### PR DESCRIPTION
**Pull request by 20DM**
* Add missing `<ctime>` includes
* Make sure Conan passes on command-line options to CMake
* Disable MPI support by default
* Add custom logger
* Remove `spdlog` dependency

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,7 +49,7 @@ as well as the **SOPT** installation:
         conan create ./cppflow/
         ```
       Note that conan requires you to specify the host (h) and the build (b) profiles on the command
-      line (`-pr:h=default -pr:b=default`), unless you have defined them in your conan profile.
+      line (`-pr=default`), unless you have defined them in your conan profile.
```
Comment (mmcleod89) [inline README.md]: I'm not sure why this changed; do we not need to set both the build and the host profiles?

**Review by mmcleod89 — COMMENTED** [review]
This looks good to me and the everything compiles and works on my machine. However I'm a bit confused as to why we aren't setting both pr:h and pr:b anymore in the README? And if this is the case then we could probably change the line above that says that we need to specify both? 

I'm also not sure why the `mpi` flag has been changed to `dompi`, although it doesn't cause any issues.

File: README.md (hunk 1 above)
Comment (20DM) [reply README.md]: I thought `-pr` takes care of both, so the wording might just need a tweak 🤔

**Comment by 20DM** [comment]
> I'm also not sure why the `mpi` flag has been changed to `dompi`, although it doesn't cause any issues.

It was`dompi` in Purify and `mpi` in Sopt - I don't care which one we use, but I do care about it being consistent 😜

File: README.md (hunk 1 above)
Comment (mmcleod89) [reply README.md]: I think that `-pr` only does the host:
```  
   -pr PROFILE_HOST, --profile PROFILE_HOST
                        Apply the specified profile to the host machine
  -pr:b PROFILE_BUILD, --profile:build PROFILE_BUILD
                        Apply the specified profile to the build machine
  -pr:h PROFILE_HOST, --profile:host PROFILE_HOST
                        Apply the specified profile to the host machine
```

(from `conan create -h`)

Comment (20DM) [reply README.md]: Ok, clarified now.

**Review by mmcleod89 — APPROVED** [review]

**Pull request merged**